filebeat/harvester/reader/encoder.go

```diff
@@ -0,0 +1,35 @@
+package reader
+
+import (
+	"io"
+	"time"
+
+	"github.com/elastic/beats/filebeat/harvester/encoding"
+)
+
+// Encode reader produces lines by reading lines from an io.Reader
+// through a decoder converting the reader it's encoding to utf-8.
+type Encode struct {
```

Comment:
We can still renamed it in combination with #2089 if something else becomes obvious.